flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NetworkEnvironmentConfiguration.java

```diff
@@ -105,6 +106,302 @@ public NettyConfig nettyConfig() {
 		return nettyConfig;
 	}
 
+	public boolean isCreditBased() {
+		return isCreditBased;
+	}
+
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Utility method to extract network related parameters from the configuration and to
+	 * sanity check them.
+	 *
+	 * @param configuration configuration object
+	 * @param maxJvmHeapMemory the maximum JVM heap size (in bytes)
+	 * @param localTaskManagerCommunication true, to skip initializing the network stack
+	 * @param taskManagerAddress identifying the IP address under which the TaskManager will be accessible
+	 * @return NetworkEnvironmentConfiguration
+	 */
+	@Deprecated
+	public static NetworkEnvironmentConfiguration fromConfiguration(
+		Configuration configuration,
+		long maxJvmHeapMemory,
+		boolean localTaskManagerCommunication,
+		InetAddress taskManagerAddress) {
+
+		// ----> hosts / ports for communication and data exchange
+
+		final int dataport = configuration.getInteger(TaskManagerOptions.DATA_PORT);
```

Comment:
Could we break down this method a bit more?
I mean adding functions:
- getDataport()
- calculateNumberOfNetworkBuffers
- createNettyConfig